### PR TITLE
[interpreter/tests] Switch to Option B'

### PR DESCRIPTION
**Pull request by rossberg**
This merges the branch with the B' design (explainer, interpreter, tests) into main.

It should be merged along with #283, which updates the spec document, and #284, which adds a new document describing the previous semantics plus respective tests.

### CODE REVIEW

File: interpreter/syntax/ast.ml [hunk 1]
```diff
@@ -185,13 +185,9 @@ and instr' =
   | Unary of unop                     (* unary numeric operator *)
   | Binary of binop                   (* binary numeric operator *)
   | Convert of cvtop                  (* conversion *)
-  | TryCatch of block_type * instr list * (* try *)
-                (var * instr list) list * (* catch exception with tag *)
-                instr list option     (* catch_all *)
-  | TryDelegate of block_type * instr list * (* try *)
-                   var                (* delegate to outer handler *)
+  | Try of block_type * catch list * instr list  (* try *)
```
Comment (titzer) [inline interpreter/syntax/ast.ml]: Do you want to keep the name `TryTable` internally to better match the text format?

Comment (rossberg) [reply interpreter/syntax/ast.ml]: Updated.

**Comment by rossberg** [comment]
Anybody willing to approve this PR?

**Review by titzer — APPROVED** [review]
lgtm, but maybe want @aheejin to also take a look before merging?

**Comment by rossberg** [comment]
Okay, I'm taking the liberty to land this for now, since some people complained, and it's not good if we keep the main doc out of date for longer. @aheejin, I'm happy to iterate in a follow-ups if you have comments later.

**Pull request merged**